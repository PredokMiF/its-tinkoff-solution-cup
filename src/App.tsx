import React from 'react'
import { RouterProvider } from 'react-router-dom'

import { router } from '~/router'

import styles from './App.module.scss'

export function App() {
    return (
        <div className={styles['app-container']}>
            <RouterProvider router={router}/>
        </div>
    )
}
