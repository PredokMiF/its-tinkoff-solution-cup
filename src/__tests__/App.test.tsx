import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'

import { App } from '~/App'

window.getComputedStyle = window.getComputedStyle || function() {}

window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    }
}

describe('App component', () => {
    test('App should exist', () => {
        const app = render(<App/>)

        expect(app).toBeTruthy()
    })

    test('App should match snapshot', () => {
        const app = render(<App/>)

        expect(app).toMatchSnapshot()
    })
})
