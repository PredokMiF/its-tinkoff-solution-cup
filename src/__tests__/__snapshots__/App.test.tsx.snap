// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > App should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_app-container_ce1e69"
      >
        <div
          class="_layout_949441"
        >
          <header
            class="_header_949441"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Главная
            </a>
            <a
              class=""
              href="/debet"
            >
              Доходы
            </a>
            <a
              class=""
              href="/credit"
            >
              Расходы
            </a>
          </header>
          <h1
            class="_title_949441"
          >
            Статистика за месяц
          </h1>
          <i>
            Загрузка...
          </i>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_app-container_ce1e69"
    >
      <div
        class="_layout_949441"
      >
        <header
          class="_header_949441"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Главная
          </a>
          <a
            class=""
            href="/debet"
          >
            Доходы
          </a>
          <a
            class=""
            href="/credit"
          >
            Расходы
          </a>
        </header>
        <h1
          class="_title_949441"
        >
          Статистика за месяц
        </h1>
        <i>
          Загрузка...
        </i>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
